spec/models/assignment_spec.rb

```diff
@@ -377,6 +377,101 @@
     end
   end
 
+  describe '#add_group_api' do
+    let(:course) { create(:course) }
+    let(:assignment) { create(:assignment, course: course) }
+    let(:user) { create(:end_user) }
+    let(:student) { create(:student, user: user, course: course) }
+    let(:user2) { create(:end_user) }
+    let(:student2) { create(:student, user: user2, course: course) }
+
+    shared_examples 'group persistence and members' do |expected_member_count: 0|
+      it 'creates persistent group and grouping' do
+        expect(group).to be_persisted
+        expect(assignment.groups).to include(group)
+        expect(group.groupings.where(assignment: assignment)).to exist
+      end
+
+      it "has #{expected_member_count} members with correct statuses" do
+        grouping = group.groupings.find_by(assignment: assignment)
+        expect(grouping.student_memberships.count).to eq(expected_member_count)
+
+        if expected_member_count > 0
+          expect(grouping.student_memberships.first.membership_status).to eq(StudentMembership::STATUSES[:inviter])
+          if expected_member_count > 1
+            expect(grouping.student_memberships.last.membership_status).to eq(StudentMembership::STATUSES[:accepted])
```

Comment:
Testing just `last` is a bit strange; it's more precise to use a loop for the rest. 

I also recommend adding an additional test case for having more than two members.